test/regridder_tests.jl

```diff
@@ -314,3 +315,54 @@ for FT in (Float32, Float64)
         end
     end
 end
+# test dataset truncation 
+@testset "test dataset truncation" begin
+    # Get the original dataset set up 
+    include(joinpath(pkgdir(ClimaCoupler), "artifacts", "artifact_funcs.jl"))
+    sst_data_all = joinpath(sst_dataset_path(), "sst.nc")
+    ds = NCDataset(sst_data_all, "r")
+    dates = ds["time"][:]
+    first_date = dates[1]
+    last_date = last(dates)
+
+    # values for the truncations 
+    time_start = 0.0
+    time_end = 1.728e6
+    # tests five cases: 
+    # 1. both start and end are earlier than in datafile 
+    # 2. start is early, end is within datafine 
+    # 3. both start and end within datafile 
+    # 4. start is in datafile end is not 
+    # 5. both start and end are later than in datafile
+    date0test = ["18690101", "18700101", "19790228", "20220301", "20230101"]
+    for date in date0test
+        date0 = DateTime(date, dateformat"yyyymmdd")
+        sst_data = Regridder.truncate_dataset(date0, time_start, time_end, sst_data_all, sst_dataset_path())
```

Comment:
We should pass in a REGRID_DIR here instead of sst_dataset_path, so the windows tests don't try to access `runneradmin/` (as discussed in our meeting)